src/Inventory/Asset/MainAsset.php

```diff
@@ -675,6 +675,12 @@ public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = 0)
             $val->states_id = $default_states_id > 0 ? $default_states_id : 0;
         }
 
+        // set states_id in known links if needed
```

Comment:
This is already done line 758. If you need it to be set earlier, maybe the existing code should be moved here?

Reply:
done